website/client/app.vue

```diff
@@ -388,7 +388,16 @@ export default {
       if (item.purchaseType === 'card') {
         this.selectedSpellToBuy = item;
 
+        // hide the dialog
         this.$root.$emit('bv::hide::modal', 'buy-modal');
+        // remove the dialog from our modal-stack,
+        // the default hidden event is delayed
+        this.$root.$emit('bv::modal::hidden', {
```

Comment:
what happens when this even is fired again by the lib and picked up by the listener in app.vue?

If the event is fired in the next tick can't we just delay the next one?

```
Vue.nextTick(() => this.$root.$emit('bv::show::modal', 'select-member-modal'));
```

Reply:
tried it right now, it still too early before the normal `hidden` event arrives. I guess bootstrap-vue uses a callback once the modal is really closed and then fires the event, which definitely takes longer than a vue-tick

Reply:
Ah okay, but just to be sure, this means the listener in app.vue will be called twice? Can this cause any problem?

Reply:
once the first `hidden` event passes, the `modalStack` is empty, and on the second call it tries to get the next modal to show again, 

but `let modalOnTop = this.$store.state.modalStack[modalStackLength - 1];` the position would be `-1` 

and `if (modalOnTop) this.$root.$emit('bv::show::modal', modalOnTop, {fromRoot: true});` doesn't do anything with an undefined `modalOnTop`

so I guess it's safe

Reply:
i'm just wondering about the case were something else is in the stack

Reply:
I can't get any other modal inside the stack, so should I add any extra checks ?

Reply:
@TheHollidayInn opinions on this?

Reply:
Talked to @negue offline, and I think this is the correct choice for now. Maybe we can resist the modal stack, but I believe we have to this for now.